src/codeu/chat/server/LocalFile.java

```diff
@@ -0,0 +1,166 @@
+package codeu.chat.server;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.io.ObjectOutputStream;
+import java.io.RandomAccessFile;
+import java.io.FileOutputStream;
+import java.io.ObjectInputStream;
+import java.io.FileInputStream;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+import codeu.chat.util.Logger;
+import codeu.chat.util.Uuid;
+
+public class LocalFile implements Serializable
+{
+    private static final long serialVersionUID = 1L;
+    //Instance varibles for saving the current data of server.
+    private final HashMap<Uuid,User> users;
+    private final HashMap<Uuid,ConversationHeader> conversationHeaders;
+    private final HashMap<Uuid,Message> messages;
+
+    private transient final File file;
+
+    private static final Logger.Log LOG = Logger.newLog(LocalFile.class);
+
+    private transient boolean hasModified = false;//It indicates if there is a new data should be handled.
+    public LocalFile(File file)
+    {
+        this.file = file;
+        LocalFile newClass;
+        HashMap<Uuid,User>tempUsers = new HashMap<>();
+        HashMap<Uuid,ConversationHeader>tempConversationHeaders= new HashMap<>();
+        HashMap<Uuid,Message>tempMessages= new HashMap<>();
+        try
+        {
+            //Deserialization
+            RandomAccessFile randomAccessFile = new RandomAccessFile(file, "r");
+            FileInputStream fileInputStream = new FileInputStream(randomAccessFile.getFD());
+		    ObjectInputStream objectInputStream = new ObjectInputStream(fileInputStream);
+            //Read the information to current instance
+		    newClass = (LocalFile)objectInputStream.readObject();
+            tempUsers = newClass.users;
+            tempConversationHeaders = newClass.conversationHeaders;
+            tempMessages = newClass.messages;
+
+            fileInputStream.close();
+            objectInputStream.close();
+        }
+        catch(Exception exception)
```

Comment:
Generally, it's not good to catch all `Exception`s in a single block like this. It's better to be specific and catch each type of exception that you expect, and react to them individually. 

In this case, the above block of code can throw an IOException and a ClassNotFoundException. If there's an IOException, that can indicate that there was a temporary error with the file system. A ClassNotFoundException on the other hand, indicates that the program is missing the classes it needs to deserialize objects.

I'd recommend using a message that indicates exactly what went wrong, and also just allowing the program to crash (see my comment below for more on that).

```java
try {

// read from file system and deserialize

} catch (IOException e) {
  System.err.println("ERROR: A transient error was encountered while reading from a local file. Please try again.");
  throw new RuntimeException(e);
  
} catch (ClassNotFoundException e) {
  System.err.println("ERROR: There was an error deserializing data. Required classes were not found.");    
  throw new RuntimeException(e);
}
```